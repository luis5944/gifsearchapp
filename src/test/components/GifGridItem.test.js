import React from "react";
const { shallow } = require("enzyme");
const { GifGridItem } = require("../../components/GifGridItem");

describe("debe mostrar el componente correctamente", () => {
  const title = "Un titulo";
  const img = "https://localhost/algo.jpg";
  const id = 1;
  const wraper = shallow(<GifGridItem id={id} title={title} img={img} />);
  test("mostrar ", () => {
    expect(wraper).toMatchSnapshot();
  });

  test("debe de tener un <p></p> con el title", () => {
    const p = wraper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("debe de tener el src img y el alt ", () => {
    const imge = wraper.find("img");

    expect(imge.props().src).toBe(img);
    expect(imge.prop("alt")).toBe(title);
  });

  test("debe  de tener una clase animate__bounce ", () => {
    const div = wraper.find("div");

    console.log(div.props().className);

    expect(div.props().className.includes("animate__bounce")).toBe(true);
  });
});
