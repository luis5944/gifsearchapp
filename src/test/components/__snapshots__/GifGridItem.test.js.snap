// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debe mostrar el componente correctamente mostrar  1`] = `
<div
  className="card animate__animated animate__bounce animate__delay-2s"
>
  <img
    alt="Un titulo"
    key="1"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
